src/api/sso/test/row-level-policies.test.js

```diff
@@ -0,0 +1,350 @@
+const { hash } = require('@senecacdot/satellite');
+const { createClient } = require('@supabase/supabase-js');
+
+const { SUPABASE_URL, ANON_KEY } = process.env;
+const {
+  ensureUsers,
+  createTelescopeUsers,
+  cleanupTelescopeUsers,
+  addFeeds,
+  cleanFeeds,
+  ensureFeeds,
+} = require('./supabaseUtil');
+const { createToken } = require('../src/token');
+
+const supabase = createClient(SUPABASE_URL, ANON_KEY);
+
+// Regular Telescope user
+const hansLippershey = {
+  firstName: 'Hans',
+  lastName: 'Lippershey',
+  email: 'hans-lippershey@example.com',
+  displayName: 'Hans Lippershey',
+  isAdmin: false,
+  isFlagged: false,
+  feeds: ['https://imaginary.blog.com/feed/hans'],
+  githubUsername: 'hlippershey',
+  githubAvatarUrl:
+    'https://avatars.githubusercontent.com/u/33902374?s=460&u=733c50a2f50ba297ed30f6b5921a511c2f43bfee&v=4',
+};
+const johannesKepler = {
+  firstName: 'Johannes',
+  lastName: 'Kepler',
+  email: 'user1@example.com',
+  displayName: 'Johannes Kepler',
+  isAdmin: false,
+  isFlagged: false,
+  feeds: ['https://imaginary.blog.com/feed/johannes'],
+  githubUsername: 'jkepler',
+  githubAvatarUrl:
+    'https://avatars.githubusercontent.com/u/7242003?s=460&u=733c50a2f50ba297ed30f6b5921a511c2f43bfee&v=4',
+};
+
+// unregistered user
+const unknowUser = {
+  firstName: 'Galileo',
+  lastName: 'Galilei',
+  email: 'user2@example.com',
+  displayName: 'Galileo Galilei',
+  isAdmin: false,
+  isFlagged: false,
+  feeds: ['https://dev.to/galileogalilei'],
+  githubUsername: 'octocat',
+  githubAvatarUrl: 'https://avatars.githubusercontent.com/u/583231?v=4',
+};
+
+// Create 2 Telescope accounts in the database
+const users = [hansLippershey, johannesKepler];
+
+// Create 2 feeds, 1 belongs to a user, 1 does not belong to anyone
+const feedList = [
+  {
+    url: hansLippershey.feeds[0],
+    wikiAuthorName: hansLippershey.displayName,
+    userId: hash(hansLippershey.email),
+  },
+  {
+    url: unknowUser.feeds[0],
+    wikiAuthorName: unknowUser.displayName,
+  },
+];
+
+describe('Unauthenticated users with public anon_key', () => {
+  describe('"telescope_profiles" security policies', () => {
+    beforeEach(async () => {
```

Comment:
Same thing here.  If you return a `Promise` to Jest, it will do the right thing.

Reply:
Good to know.